### ISSUE TITLE
Avoid having to transpile for Node.js

### ISSUE THREAD

**Issue by philidem** [andrejewski/n-level-cache#5]
If you removed the usage of `export default` from `index.js` then we wouldn't need to use the babel transformed version of this module.

For ultimate compatibility and simplicity can you change the JavaScript code to avoid `import`/`export`?

I also think it is bad practice to export a single class because it makes your JavaScript API more rigid. I would prefer something like `require('n-level-cache').create(options)`.

For browsers, I am ok with using transpiling. You could also minimize the amount of generated code by _not_ using the `class` syntax.

The `class` syntax adds this non-trivial generated code:

``` javascript
var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
```

In summary:
- Use `module.exports` or `exports.create = function() {...}` instead of `export default` so that we don't have to transpile for Node.js
- Change the JavaScript API to not export a class and instead use `require('n-level-cache').create(...)`
- Don't use `class` keyword because it doesn't add much value here and it avoids extra generated code bloat (helpful if targeting the browser runtime environment)

**Comment by andrejewski** [comment]
I will replace `export default` with `module.exports =`. You can then work with `require('n-level-cache/src')`. I don't agree about `create()`, being able to use `instanceof` and `extends` is more important to me.

**Comment by philidem** [comment]
Thanks for removing `export default` (that's what I care most about)!

Using `.create(...)` doesn't prevent you from using `instanceof` or `extends`; you would just expose it a little differently... For example, you could expose `require('n-level-cache).extend(...)` but in JavaScript world we typically avoid some of those Object Oriented patterns. Anyways, not a big deal to me. I just want to be able to use your module on server-side with no transpiling :)

Thanks for your help on this!

**Comment by andrejewski** [comment]
Fixed in 585c3f8a53b2bbf4a88cf7aa47e134b2d6dde48a

**Comment by philidem** [comment]
I'm only slightly offended that your commit referred to "legacy projects". We're nearly using the latest Node.js :)

Thanks for fix!
